plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -405,7 +405,21 @@ hoistExpr var t =
                     (PIR.Def var' (PIR.mkVar () var', PIR.Strict))
                     mempty
 
-                t' <- compileExpr t
+                CompileContext {ccOpts=profileOpts} <- ask
+                t' <- do
+                    if coProfile profileOpts==All then do
+                        let ty = PLC.varDeclType var'
+                        t'' <- compileExpr t
+                        let tInText = T.pack (show t'')
+                            -- Put "exiting" in the body
+                            traceInside (LamAbs () n x body) = LamAbs () n x (traceInside body)
+                            traceInside e                    = mkTrace ty ("exiting"<> tInText) e
+                        return $
+                            mkTrace
```

Comment:
This isn't quite right, I think. This puts the trace around the whole lambda, so it will trigger when we evaluate the lambda itself, not when we apply it. I think we need to do what I wrote in the ticket, something like
```
\a -> \b -> (trace @(() -> c) "entering f" (\() -> trace @c "exiting f" body) ())
```

Reply:
```
\a -> \b -> FORCE (trace "entering f" (DELAY (trace "exiting f" body)))
```

Reply:
```
let f = x +1
in f 1 + f 2
```